docs/snapshot.md

```diff
@@ -0,0 +1,22 @@
+Snapshot Design
+==================
```

Comment:
In addition to sequence diagrams, could you also write down the design in text like [this](https://github.com/topolvm/topolvm/blob/main/docs/design.md)?